src/sentry/static/sentry/app/views/eventsV2/saveQueryButton.tsx

```diff
@@ -46,9 +46,8 @@ class EventsSaveQueryButton extends React.Component<Props, State> {
     ) {
       // eslint-disable-next-line react/no-did-update-set-state
       this.setState({queryName: this.props.eventView.name || ''});
-    }
-    // entering or leaving edit mode
-    if (this.props.isEditing !== prevProps.isEditing) {
+    } else if (this.props.isEditing !== prevProps.isEditing) {
```

Comment:
@markstory looks like you can make this into one if condition; by adding `this.props.isEditing !== prevProps.isEditing` as an or condition.